test/run-tests

```diff
@@ -707,23 +680,22 @@ def handle_task(gh, args, config, task, ansible_id, dry_run=False):
         # we wake up, we know that nobody else wants to do the same task and
         # can go ahead. Otherwise, choose something else.
 
-        gh.post(
-            f"repos/{task.owner}/{task.repo}/statuses/{task.head}",
-            {"context": status_context, "state": "pending", "description": description},
+        commit = gh.get_repo(f"{task.owner}/{task.repo}").get_commit(task.head)
+        commit.create_status(
+            state="pending", context=status_context, description=description
```

Comment:
Note that this now takes 2 requests instead of 1 - I'm assuming it is doing a request to get the commit, then another request to create the status

Reply:
maybe 3 requests - `get_repo` then `get_commit` then `create_status`

Reply:
Can we stash objects such as commit in Task?

Reply:
yes